packages/@ember/-internals/glimmer/tests/integration/components/tracked-test.js

```diff
@@ -44,6 +45,27 @@ if (EMBER_METAL_TRACKED_PROPERTIES) {
         this.assertText('max jackson | max jackson');
       }
 
+      '@test returning new promise array proxies does not cause perpetual rerenders'() {
```

Comment:
I'd recommend renaming this.  This PR fixes `ArrayProxy` specifically, but the underlying problem is still there for promise array proxies (or any non-memoized getter that reads a `@tracked` value that causes it to be later mutated)

Maybe
```js
'@test returning ArrayProxy in un-memoized getter does not cause perpetual rerenders'
```

Reply:
@pzuraq - I think this is the only remaining issue to address before landing...